addon/content/reducer/reducer-summary.js

```diff
@@ -49,7 +50,10 @@ async function handleShowDetails(messages, state, dispatch, updateFn) {
 export const summaryActions = {
   replaceConversation({ summary, messages }) {
     return async (dispatch, getState) => {
-      await handleShowDetails(messages, getState(), dispatch, () => {
+      await handleShowDetails(messages, getState(), dispatch, async () => {
+        // The messages inside `msgData` don't come with filled in `to`/`from`/ect. fields.
+        // We need to fill them in ourselves.
+        await mergeContactDetails(messages.msgData);
```

Comment:
As demonstrated in #1620, I think we need to make this not await - so we fire the request and handle the response if/when we get it.

I'm happy to not address that here, as long as we're happy the architecture will support it, but I think we should aim to do it soon.

Reply:
I think if we put a per-conversation email cahce in Redux and use a reducer to create the formatted emails, then the reducer can have a nice fallback and the rest of the details will show up with they're available. This should eliminate the performance problems.

Reply:
I'd be fine with that, though if we can do it per `stub.xhtml` display rather than per conversation, that might be even better - as we'd then cache emails across conversations.